src/transport_picoquic.cpp

```diff
@@ -766,26 +781,30 @@ PicoQuicTransport::sendTxData(StreamContext* stream_cnx, [[maybe_unused]] uint8_
         if (max_len >= out_data.value().size()) {
             metrics.dgram_sent++;
 
-            uint8_t* buf = NULL;
+            uint8_t* buf = nullptr;
 
             if (stream_cnx->stream_id == 0) {
                 buf = picoquic_provide_datagram_buffer_ex(bytes_ctx,
                                                           out_data.value().size(),
-                                                          /*stream_cnx->tx_data.size() > 1 ? 1 : 0*/ 1);
+                                                          1);
 
             } else {
                 buf = picoquic_provide_stream_data_buffer(bytes_ctx,
                                                           out_data->size(),
                                                           0,
-                                                          /*stream_cnx->tx_data.size() > 1 ? 1 : 0*/ 1);
+                                                          1);
             }
 
-            if (buf != NULL) {
+            if (buf != nullptr) {
                 std::memcpy(buf, out_data.value().data(), out_data.value().size());
-
                 stream_cnx->tx_data->pop();
             }
         }
+    } else {
+        // has_value false
+        //std::cout << "send: picoquic_provide_datagram_buffer_ex masrking inactive" << std::endl;
+
+        picoquic_provide_datagram_buffer_ex(bytes_ctx, 0, 0);
```

Comment:
Instead of adding an else clause, why not return early if there's no value? That would reduce the amount of tabbing in the body of the function, and remove the need for the comment inside this statement.

Reply:
we can return early since we need to let the stack know .. that was the one of the reasons for busy loop earlier